tests/aws/aws_environment.rs

```diff
@@ -74,6 +79,97 @@ fn deploy_a_working_environment_with_no_router_on_aws_eks() {
+    let test_name = function_name!();
+    engine_run_test(|| {
+        init();
+        let span = span!(Level::INFO, "test", name = test_name);
+        let _enter = span.enter();
+
+        let logger = logger();
+        let secrets = FuncTestsSecrets::new();
+        let context = context(
+            secrets
+                .AWS_TEST_ORGANIZATION_ID
+                .as_ref()
+                .expect("AWS_TEST_ORGANIZATION_ID is not set")
+                .as_str(),
+            secrets
+                .AWS_TEST_CLUSTER_ID
+                .as_ref()
+                .expect("AWS_TEST_CLUSTER_ID is not set")
+                .as_str(),
+        );
+
+        let mut environment = test_utilities::common::working_minimal_environment(
+            &context,
+            secrets
+                .DEFAULT_TEST_DOMAIN
+                .expect("DEFAULT_TEST_DOMAIN is not set in secrets")
+                .as_str(),
+        );
+
+        let ecr = container_registry_ecr(&context);
+        let local_docker = build_platform_local_docker(&context);
+        let app = environment.applications.first().unwrap();
+        let image = app.to_image();
+
+        let app_build = app.to_build();
+        let _ = match local_docker.has_cache(&app_build) {
+            Ok(CacheResult::Hit) => assert!(false),
+            Ok(CacheResult::Miss(parent_build)) => assert!(true),
+            Ok(CacheResult::MissWithoutParentBuild) => assert!(false),
+            Err(err) => assert!(false),
+        };
+
+        let _ = match ecr.pull(&image).unwrap() {
+            PullResult::Some(_) => assert!(false),
+            PullResult::None => assert!(true),
+        };
+
+        let build_result = local_docker.build(app.to_build(), false).unwrap();
+
+        let _ = match ecr.push(&build_result.build.image, false) {
+            Ok(_) => assert!(true),
+            Err(_) => assert!(false),
+        };
+
+        // TODO clean local docker cache
+
+        let start_pull_time = SystemTime::now();
+        let _ = match ecr.pull(&build_result.build.image).unwrap() {
+            PullResult::Some(_) => assert!(true),
+            PullResult::None => assert!(false),
+        };
+
+        let pull_duration = SystemTime::now().duration_since(start_pull_time).unwrap();
+
+        let _ = match local_docker.has_cache(&build_result.build) {
+            Ok(CacheResult::Hit) => assert!(true),
+            Ok(CacheResult::Miss(parent_build)) => assert!(false),
+            Ok(CacheResult::MissWithoutParentBuild) => assert!(false),
+            Err(err) => assert!(false),
+        };
+
+        let start_pull_time = SystemTime::now();
+        let _ = match ecr.pull(&image).unwrap() {
+            PullResult::Some(_) => assert!(true),
+            PullResult::None => assert!(false),
+        };
+
+        let pull_duration_2 = SystemTime::now().duration_since(start_pull_time).unwrap();
+
+        if pull_duration_2.as_millis() > pull_duration.as_millis() {
```

Comment:
Hopefully I won't be flacky, also, maybe, once we got an idea of the % of speed up it brings, maybe worth adding it here. 
Because now, even if it's 1 milli quicker, it works, but maybe that's not what we want.